### PR TITLE
Get-DbaDatabase / Compare-DbaCollationSensitiveObject - Removed mandatory from value parameter

### LINKED ISSUE #8225: Issue with Get-DbaDatabase -NoFullBackup and Compare-DbaCollationSensitiveObject
### Verified issue does not already exist?

Yes

### What error did you receive?

Compare-DbaCollationSensitiveObject : Cannot bind argument to parameter 'Value' because it is null.
At C:\Program Files\WindowsPowerShell\Modules\dbatools\1.1.65\allcommands.ps1:22813 char:109
+ ... bject -Property Name -In -Value $lastCopyOnlyBackups.Database -Collat ...
+                                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : InvalidData: (:) [Compare-DbaCollationSensitiveObject], ParameterBindingValidationExcept
   ion
    + FullyQualifiedErrorId : ParameterArgumentValidationErrorNullNotAllowed,Compare-DbaCollationSensitiveObject

### Steps to Reproduce

```powershell
# provide your command(s) executed pertaining to dbatools
# please include variable values (redacted or fake if needed) for reference
Get-DbaDatabase -sqlinstance server01 -NoFullBackup
```

### Are you running the latest release?

Yes

### Other details or mentions

I have run this command on Server 2012 R2 and Server 2019 with the same errors. PowerShell 5.1 versions and PowerShell 7.2 give the same error.

dbatools 1.1.36 works, but doesn't use the Compare-DbaCollationSensitiveObject function.

### What PowerShell host was used when producing this error

PowerShell Core (pwsh.exe), Windows PowerShell (powershell.exe), Windows PowerShell ISE (powershell_ise.exe)

### PowerShell Host Version

2012 R2 Server:
Name                           Value                                                                                                             
----                           -----                                                                                                             
PSVersion                      5.1.14409.1005                                                                                                    
PSEdition                      Desktop                                                                                                           
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                           
BuildVersion                   10.0.14409.1005                                                                                                   
CLRVersion                     4.0.30319.42000                                                                                                   
WSManStackVersion              3.0                                                                                                               
PSRemotingProtocolVersion      2.3                                                                                                               
SerializationVersion           1.1.0.1                                                                                                           

2019 Server:
Name                           Value                                                                                                                                     
----                           -----                                                                                                                                     
PSVersion                      5.1.17763.2268                                                                                                                            
PSEdition                      Desktop                                                                                                                                   
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                   
BuildVersion                   10.0.17763.2268                                                                                                                           
CLRVersion                     4.0.30319.42000                                                                                                                           
WSManStackVersion              3.0                                                                                                                                       
PSRemotingProtocolVersion      2.3                                                                                                                                       
SerializationVersion           1.1.0.1                                                                                                                                   

### SQL Server Edition and Build number

Multiple versions and patch levels ranging from 2008 R2 through 2019 CU15.

### .NET Framework Version

2012 R2 Server:
.NET Framework 4.8.3928.0

2019 Server:
.NET Framework 4.8.4420.0

### PR DESCRIPTION
**Pull request by ACALVETT**
## Type of Change

 - [X] Bug fix (non-breaking change, fixes #8225  )
 - [ ] New feature (non-breaking change, adds functionality, fixes # )
 - [ ] Breaking change (effects multiple commands or functionality, fixes # )
 - [ ] Ran manual Pester test and has passed (`.\tests\manual.pester.ps1`)
 - [ ] Adding code coverage to existing functionality
 - [x] Pester test is included
 - [ ] If new file reference added for test, has is been added to github.com/dataplat/appveyor-lab ?
 - [ ] Unit test is included
 - [ ] Documentation
 - [ ] Build system

### Purpose
Prevent null error when no full backups returned

### Approach
Compare-DbaCollationSensitiveObject now allows null and returns a result set. I think this is a valid approach over testing for $null before calling the function as you still need the results of the null comparison.

Function is not used anywhere else making this a safe change

### CODE REVIEW

**Comment by ACALVETT** [comment]
I don't think i can fix the failing tests because its failing on below command saying the CLIENTSECRET is empty, can someone help?

```
       $securestring = ConvertTo-SecureString $env:CLIENTSECRET -AsPlainText -Force
```

**Comment by wsmelton** [comment]
Can you also add a test for this functionality in `Get-DbaDatabase.Tests.ps1`? This would have caught the initial issue in the previous PR.

**Comment by ACALVETT** [comment]
Sure, will take a look and put a test together.

**Comment by ACALVETT** [comment]
First test i have ever written, basic but i think it meets the requirements and from what i can tell i think its working.

**Comment by andreasjordan** [comment]
Failing test says: Get-DbaDatabase.Tests.ps1 has no trailing spaces (line(s) 48)

**Comment by ACALVETT** [comment]
Fixed, how do i see that in the logs?

**Comment by andreasjordan** [comment]
I think only some maintainers can see those logs - that's why I have a look and post the messages.

**Comment by andreasjordan** [comment]
Now there are only problems related to the build reference file. You can push an empty commit with "(do Get-DbaDatabase)" to limit the testing scope.

File: tests/Get-DbaDatabase.Tests.ps1 [hunk 1]
```diff
@@ -35,6 +35,32 @@ Describe "$CommandName Integration Tests" -Tags "IntegrationTests" {
...
+
+}
+
+Describe "$CommandName Integration Tests" -Tag "IntegrationTests" {
+    BeforeAll {
+
+        $random = Get-Random
+        $dbname1 = "dbatoolsci_Backup_$random"
+        $dbname2 = "dbatoolsci_NoBackup_$random"
+        New-DbaDatabase -SqlInstance $script:instance1 -name $dbname1 , $dbname2
```
Comment (andreasjordan) [inline tests/Get-DbaDatabase.Tests.ps1:47]: Maybe add `$null = ` to suppress the output of `New-DbaDatabase` (and change `-name ` to `-Name`?)

**Comment by andreasjordan** [comment]
Change looks good to me.

**Comment by potatoqualitee** [comment]
Thanks everyone 🥳

**Comment by wsmelton** [comment]
> how do i see/navigate the logs

@ACALVETT you can access this via the checks tab of your PR. Everyone can see our Appveyor builds.

They are also accessible going directly to: https://ci.appveyor.com/project/dataplat/dbatools (this list all jobs for branches and PR).

File: tests/Get-DbaDatabase.Tests.ps1 (hunk 1 above)
Comment (ACALVETT) [reply tests/Get-DbaDatabase.Tests.ps1:47]: Change makes sense to me, should i open a new PR for it?

Comment (andreasjordan) [reply tests/Get-DbaDatabase.Tests.ps1:47]: Just if you have time...

**Pull request merged**